nevergrad/functions/pownet/solver.py

```diff
@@ -0,0 +1,251 @@
+    elif force_update:
+        git.cmd.Git(dataset_path).pull('origin', 'main') #Check update
+    if dataset_name == "cambodian":
+        return os.path.join(dataset_path, 'pypownetr', 'datasets', 'kamal0013', 'camb_2016')
+    else:
+        raise NotImplementedError
+
+def _load_sim_data(dst, src, nodes, horizon_hours, day):
+    for z in nodes:
+        for i in range(1, horizon_hours+1):
+            dst[z,i] = src[z,(day-1)*24+i]
+
+
+def solve_pownet_subproblems(pyomo_model, model_data_path, solver, year=2016, start_day=1, last_day=365):
+    """
+    simulation year, start(1-365) and end(1-365) days of simulation
+    """
+    
+    instance = pyomo_model.create_instance(model_data_path)
+
+    ###solver and number of threads to use for simulation
+    H = instance.HorizonHours
+    K = range(1, H+1)
+
+    ###Run simulation and save outputs
+    #Containers to store results
+    on = []
+    switch = []
+
+    mwh = []
+    hydro = []
+    solar = []
+    wind = []
+
+    hydro_import = []
+
+    srsv = []
+    nrsv = []
+
+    vlt_angle = []
+
+    system_cost = []
+
+    for day in range(start_day, last_day+1):
+        if hasattr(instance, 'd_nodes'):
+            for z in instance.d_nodes:
+                # Load Demand and Reserve time series data
+                for i in K:
+                    instance.HorizonDemand[z, i] = instance.SimDemand[z, (day-1)*24+i]
+                    instance.HorizonReserves[i] = instance.SimReserves[(day-1)*24+i] 
+                
+        if hasattr(instance, 'h_nodes'):
+            for z in instance.h_nodes:
+                # Load Hydropower time series data
+                for i in K:
+                    instance.HorizonHydro[z, i] = instance.SimHydro[z, (day-1)*24+i]
+                
+        if hasattr(instance, 's_nodes'):
+            for z in instance.s_nodes:
+                # Load Solar time series data
+                for i in K:
+                    instance.HorizonSolar[z, i] = instance.SimSolar[z, (day-1)*24+i]
+                
+        if hasattr(instance, 'w_nodes'):
+            # _load_sim_data(instance.HorizonWind, instance.SimWind, instance.w_nodes, H, day)
+            for z in instance.w_nodes:
+                # Load Wind time series data
+                for i in K:
+                    instance.HorizonWind[z, i] = instance.SimWind[z, (day-1)*24+i]
+                
+        if hasattr(instance, 'h_imports'):
+            
+            for z in instance.h_imports:
+                # Load Hydropower time series data
+                for i in K:
+                    instance.HorizonHydroImport[z,i] = instance.SimHydroImport[z,(day-1)*24+i]     
+        
+        func = core.Pyomo(instance)
+        optimizer = ng.optimizers.CMA(parametrization=func.parametrization, budget=10)
+        recommendation = optimizer.minimize(func.function)
+        print(recommendation) #Low system_cost is better
```

Comment:
I would avoid prints in the code (and commented section, but then again it's WIP is that it?)